packages/mui-joy/src/styles/extendTheme.ts

```diff
@@ -644,5 +645,38 @@ export default function extendTheme(themeOptions?: CssVarsThemeOptions): Theme {
     });
   };
 
+  // @ts-ignore palette could be defined in the ThemeProvider's theme
+  theme.palette = deepmerge(themeOptions?.palette || {}, {
+    ...(theme.colorSchemes.light.palette as RuntimeColorSystem['palette']),
+    colorScheme: 'light',
+  });
+  theme.vars = {
+    focus: theme.focus,
+    fontFamily: theme.fontFamily,
+    fontSize: theme.fontSize,
+    fontWeight: theme.fontWeight,
+    letterSpacing: theme.letterSpacing,
+    lineHeight: theme.lineHeight,
+    radius: theme.radius,
+    shadow: theme.shadow,
+    palette: theme.palette,
+    shadowRing: theme.shadowRing,
+    shadowChannel: theme.shadowChannel,
+  };
```

Comment:
Should `theme.vars` comes from `cssVarsParser`?

```js
import cssVarsParser from './cssVarsParser';

...
const { vars } = cssVarsParser(theme, { prefix: cssVarPrefix, shouldSkipGeneratingVar });

theme.vars = vars;
```

Because `theme.vars.*` should return a CSS variable:

```js
const theme = extendTheme();
console.log(theme.vars.palette.primary[100]) // 'var(--joy-palette-primary-100)'
```

Reply:
Unless I am missing something, without using a `CssVarsProvider` none of these variables are available, as they are not added in any stylesheet yet. This is why I just copied the actual CSS values in the `vars` object. Otherwise, the default value would point out to non existing CSS vars.

Reply:
We decided the variables generated in extendTheme to contain default values.